### PR TITLE
Motorized Wheelchair improvements

### PR DESCRIPTION
**Pull request by lizardqueenlexi**
## About The Pull Request

This PR improves motorized wheelchairs in the following ways:
- On initialize, motorized wheelchairs will have the most basic possible stock parts inserted. This resolves a bug where motorized wheelchairs spawned by admins or other means (mail, mapping) were completely worthless, as they had no stock parts in them and you could not insert better ones. When crafted, these basic parts are deleted and replaced with the correct ones.
- Power cells are now part of the motorized wheelchair recipe, instead of being inserted after.
- When deconstructed or destroyed, motorized wheelchairs now drop their power cell on the floor with everything else instead of deleting it from existence.
- Miscellaneous old code has been tidied up, removing one-letter var names, pointless code duplication, and the like.
## Why It's Good For The Game

I've been talking to Dalm about this, and wheelchairs are in dire need of some improvements. Tidying up the mess that is the motorized wheelchair seemed like a good place to start.

Firstly, it's good for motorized wheelchairs to function as expected when not specifically crafted by hand. This is helpful to admins, mappers, and anyone else who might have reason to make one of these appear. It's also good that all the parts in the wheelchair drop on the ground when destroyed, as there's no particular reason power cells should be lost.

As for power cells being included in the recipe - it's not obvious at a glance that you _need_ to insert a power cell after crafting, until you try to drive your new wheelchair around. This extra step of annoyance is now avoided by making it required to stick one in when you build the thing.
## Changelog
:cl:
fix: Motorized wheelchairs will no longer spawn in a bugged state where they have no parts and can't be upgraded.
fix: Motorized wheelchairs will drop their power cell when destroyed or deconstructed.
qol: Power cells are now inserted into motorized wheelchairs as part of the crafting recipe, instead of as an extra step afterwards.
/:cl:

### CODE REVIEW

File: code/modules/vehicles/motorized_wheelchair.dm [hunk 1]
```diff
@@ -114,8 +135,8 @@
 		if(istype(newstockpart, type_to_check) && istype(oldstockpart, type_to_check))
 			if(newstockpart.tier > oldstockpart.tier)
 				// delete the part in the users hand and add the datum part to the component_list
-				I.moveToNullspace()
-				qdel(I)
+				attacking_item.moveToNullspace()
```
Comment (SyncIt21) [inline code/modules/vehicles/motorized_wheelchair.dm:138]:
```suggestion
```
Deleting an item already moves it to null space, This is redundant

**Review by SyncIt21 — COMMENTED** [review]
More nitpicks

**Comment by Dawnseer** [comment]
With the cell being part of the recipe, does it take into account how charged the cell is? And can you still freely take out the cell and replace it like before?

**Comment by lizardqueenlexi** [comment]
> With the cell being part of the recipe, does it take into account how charged the cell is? And can you still freely take out the cell and replace it like before?

Yes to both. Whatever power cell you use to craft it is the same as what ends up inside of it, and you can still pull it out as always.

File: code/modules/vehicles/wheelchair.dm [hunk 2]
```diff
@@ -52,15 +52,16 @@
...
-	to_chat(user, span_notice("You begin to detach the wheels..."))
-	if(I.use_tool(src, user, 40, volume=50))
-		to_chat(user, span_notice("You detach the wheels and deconstruct the chair."))
-		new /obj/item/stack/rods(drop_location(), 6)
-		new /obj/item/stack/sheet/iron(drop_location(), 4)
-		qdel(src)
-	return TRUE
+	to_chat(user, span_notice("You begin to disassemble the wheelchair..."))
+	if(!tool.use_tool(src, user, 4 SECONDS, volume=50))
+		return
```
Comment (JohnFulpWillard) [inline code/modules/vehicles/wheelchair.dm:59]: should this not return ``TOOL_ACT_TOOLTYPE_SUCCESS``?

File: code/modules/vehicles/wheelchair.dm [hunk 3]
```diff
@@ -52,15 +52,16 @@
...
+/obj/vehicle/ridden/wheelchair/wrench_act(mob/living/user, obj/item/tool) //Attackby should stop it attacking the wheelchair after moving away during decon
 	..()
-	to_chat(user, span_notice("You begin to detach the wheels..."))
-	if(I.use_tool(src, user, 40, volume=50))
-		to_chat(user, span_notice("You detach the wheels and deconstruct the chair."))
-		new /obj/item/stack/rods(drop_location(), 6)
-		new /obj/item/stack/sheet/iron(drop_location(), 4)
-		qdel(src)
-	return TRUE
+	to_chat(user, span_notice("You begin to disassemble the wheelchair..."))
```
Comment (JohnFulpWillard) [inline code/modules/vehicles/wheelchair.dm:57]: would this be better as balloon alerts?

File: code/modules/vehicles/motorized_wheelchair.dm [hunk 4]
```diff
@@ -126,17 +141,26 @@
 				break
 	refresh_parts()
 
-/obj/vehicle/ridden/wheelchair/motorized/wrench_act(mob/living/user, obj/item/I)
-	to_chat(user, span_notice("You begin to detach the wheels..."))
-	if(!I.use_tool(src, user, 40, volume=50))
-		return TRUE
+/obj/vehicle/ridden/wheelchair/motorized/wrench_act(mob/living/user, obj/item/tool)
+	to_chat(user, span_notice("You begin to disassemble the wheelchair..."))
```
Comment (JohnFulpWillard) [inline code/modules/vehicles/motorized_wheelchair.dm:145]: Same thing here

**Comment by lizardqueenlexi** [comment]
wait what the fuck is going on with these CI tests

**Comment by SyncIt21** [comment]
Screenshot_20231215-144639_Chrome

Replace it TOOL_ACT_TOOLTYPE_SUCCESS with
ITEM_INTERACT_SUCCESS

You've been gone a while during which the entire attack chain got refactored

**Review by Ghommie — APPROVED** [review]

**Pull request merged**